text/0178-mapping-to-otlp-anyvalue.md

```diff
@@ -0,0 +1,227 @@
+is capable of representing primitive and structured data of certain types.
+
+Implementations that have a source data in any form, such as in-memory objects
+or data coming from other formats that needs to be converted to AnyValue SHOULD
+follow the rules described below.
+
+### Primitive Values
+
+#### Integer Values
+
+Integer values which are within the range of 64 bit signed numbers
+[-2^63..2^63-1] SHOULD be converted to AnyValue's
+[int_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L33)
+field.
+
+Integer values which are outside the range of 64 bit signed numbers SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field using decimal representation.
+
+#### Enumerations
+
+Values, which belong to a limited enumerated set (e.g. a Java
+[enum](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)) SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field with the value of the string set to the symbolic name of the enumeration.
+
+If the symbolic name of the enumeration is not possible to obtain the
+implementation SHOULD map enumeration's value to AnyValue's
+[int_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L33)
+field set equal to enum's ordinal number when such ordinal number is naturally
+obtainable.
+
+If the ordinal value is also not possible to obtain the enumeration SHOULD be
+converted to AnyValue's
+[bytes_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L37)
+field in any manner that the implementation deems reasonable.
+
+#### Floating Point Values
+
+Floating point values which are within the range and precision of IEEE 754
+64-bit floating point numbers (including IEEE 32-bit floating point values)
+SHOULD be converted to AnyValue's
+[double_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L34)
+field.
+
+Floating point values which are outside the range or precision of IEEE 754
+64-bit floating point numbers (e.g. IEEE 128-bit floating bit values) SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
+field using decimal floating point representation.
+
+#### String Values
+
+String values which are valid Unicode sequences SHOULD be converted to
```

Comment:
Does an encoding of Unicode need to be specified?

Reply:
I don't think so. OTLP Protobuf requires UTF8, but the source from which we map can be any other Unicode encoding and can be converted to UTF8. So I think saying "valid Unicode" is a sufficient restruction.